test/e2e/operator_test.go

```diff
@@ -159,6 +160,14 @@ func TestMain(m *testing.M) {
 		fmt.Printf("Failed to create rolebinding external-dns-operator in ns %s: %v\n", operandNamespace, err)
 	}
 
+	if err = ensureOperandClusterRoleBinding(); err != nil {
+		fmt.Printf("Failed to create clusterrolebinding external-dns : %v\n", err)
```

Comment:
I think you can pass t *testing.T in the function and then add t.Errorf rather than using fmt.Printf

Reply:
Passing in `t` and to the test helper and then using `t.Helper()`, `t.Errorf()` etc. inside the helper is generally good practice.  However, in this case, the helper is used by the test runner which calls `m.Run()`, so I don't think there is a `t` that could be passed in to the helpers.

Reply:
Thanks @Miciah

Reply:
@Miciah  @sherine-k   OK I am not sure what you mean but I can here it is added https://github.com/sherine-k/external-dns-operator/blob/7df0b56243b1fdf04e3bf6121a24d2f4eae76819/test/e2e/operator_test.go#L191

here Fatalf is used https://github.com/sherine-k/external-dns-operator/blob/7df0b56243b1fdf04e3bf6121a24d2f4eae76819/test/e2e/operator_test.go#L237

I think we shall exit and not just Printf and error

Reply:
Regarding `t`, here's a simple question:  What `t`?  

Calling `os.Exit(1)` if the ensure function returns a non-nil error value might make sense.  We can also introduce a polling loop around the create operation if we need to make the test more resilient.  I wouldn't worry too much though unless we see the ensure steps fail.

Reply:
Hmm OK whatever you say.

Reply:
Oh OK BTW I am very sorry I just saw that this function is present in TestMain function which has *testing.M